### PR TITLE
Spells/Priest: Renewed Hope also affects Flash Heal

### PR DESCRIPTION
**Pull request by ghost**
by @P-Kito; retrieved from #15027

The talent Renewed Hope (2 ranks) should also affect spell 2061, Flash Heal (all ranks).
DBC is missing Flash Heal in Spell Overrides.

### CODE REVIEW

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -10532,6 +10532,13 @@ float Unit::GetUnitSpellCriticalChance(Unit* victim, SpellInfo const* spellProto
                 // Custom crit by class
                 switch (spellProto->SpellFamilyName)
                 {
+                    case SPELLFAMILY_PRIEST:
+                        // Renewed Hope should affect flash heal
+                        if (spellProto->SpellFamilyFlags[0] & 0x800 && spellProto->SpellIconID == 177)
```
Comment (Carbenium) [inline src/server/game/Entities/Unit/Unit.cpp]: Are you sure this check is correct?
This selects 6788 - Weakened Soul but not Flash Heal

Comment (P-Kito) [inline src/server/game/Entities/Unit/Unit.cpp]: .... Can you please read the spell RENEWED HOPE.
http://wotlk.openwow.com/spell=57472

Comment (ccrs) [inline src/server/game/Entities/Unit/Unit.cpp]: dont trust openwow or wowhead on dbc data, use spellwork.

Comment (P-Kito) [inline src/server/game/Entities/Unit/Unit.cpp]: Wtf do you want now? We are not even talking about any effects, basepoints or whatever................... just what it does in general... and ofc I use spellwork for those kind of information.

**Comment by joschiwald** [comment]
i think in this case it's better to hack dbc values

```c++
case 57470: // Renewed Hope (Rank 1)
case 57472: // Renewed Hope (Rank 2)
    // also affect Flash heal
    spellInfo->Effect[EFFECT_0].SpellClassMask[0] |= 0x800;
    break;
```

File: src/server/game/Entities/Unit/Unit.cpp (hunk 1 above)
Comment (ccrs) [inline src/server/game/Entities/Unit/Unit.cpp]: shout? O.o

**Comment by P-Kito** [comment]
Oh I didnt know you can do that.

**Comment by Frytiks** [comment]
spellInfo->Effect[EFFECT_0] -> should be spellInfo->Effects[EFFECT_0]

**Comment by tkrokli** [comment]
@joschiwald, @P-Kito : seems like a good idea to skip the if-testing. BTW; is Frytiks' comment correct?
`spellInfo->Effect[EFFECT_0]` should be `spellInfo->Effects[EFFECT_0]` (plural form `Effects`)

**Comment by Frytiks** [comment]
Yes it's correct

**Comment by tkrokli** [comment]
Sorry about the delay, I got distracted with other work. Making the suggested change now. :memo:

**Comment by lendan** [comment]
@tkrokli hit me up on irc ill tell you what you did wrong

**Comment by Aokromes** [comment]
Some checks have failed — The Travis CI build failed

**Comment by tkrokli** [comment]
Yes, I know. Spoke with Nayd and Riztazz about that in irc. Trying to replace the commit with correct fix.

**Comment by lendan** [comment]
@tkrokli
Cmon mate,come on irc for few mins

**Pull request closed without merging**